metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -117,11 +116,11 @@ class Compilers(
   def completionItemResolve(
       item: CompletionItem,
       token: CancelToken
-  ): Option[CompletionItem] = {
+  ): Option[Future[CompletionItem]] = {
```

Comment:
Is there any way we can make make it `Future[Option[_]]` ? It seems a more natural way of working with futures and will avoid Future.successful(null) later on. 

What we would really need is cats here haha.

Reply:
I agree it's generally desirable to keep the `Future` at the outermost layer. This particular API is only used by `MetalsLanguageServer` where the results go straight to lsp4j so I was lazy and returned whatever shape came out of the implementation.

Reply:
I am fine with that., since it's better we don't add more not needed dependencies.

What I meant with cats was this : https://typelevel.org/cats/datatypes/optiont.html